src/modules/functionality.js

```diff
@@ -0,0 +1,130 @@
+const Display = () => {
+  // SELECT ELEMENTS
+  const form = document.getElementById('todoform');
+  const todoInput = document.getElementById('newtodo');
+  const todosListEl = document.getElementById('todos-list');
+  // const notificationEl = document.querySelector('.notification');
+
+  // VARS
+  let todos = JSON.parse(localStorage.getItem('todos')) || [];
+  let EditTodoId = -1;
+
+  // 1st render
+  renderTodos();
+
+  // FORM SUBMIT
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+
+    saveTodo();
+    renderTodos();
+    localStorage.setItem('todos', JSON.stringify(todos));
+  });
+
+  // SAVE TODO
+  let saveTodo = () => {
+    const todoValue = todoInput.value;
+
+    // check if the todo is empty
+    const isEmpty = todoValue === '';
+
+    // check for duplicate todos
+    const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+
+    if (isEmpty) {
+      showNotification("Todo's input is empty");
+    } else if (isDuplicate) {
+      showNotification('Todo already exists!');
+    } else {
+      if (EditTodoId >= 0) {
+        todos = todos.map((todo, index) => ({
+          ...todo,
+          value: index === EditTodoId ? todoValue : todo.value,
+        }));
+        EditTodoId = -1;
+      } else {
+        todos.push({
+          value: todoValue,
+          checked: false,
+          color: `#${Math.floor(Math.random() * 16777215).toString(16)}`,
+        });
```

Comment:
I love the way you have styled the application and customized the check box, but each task should have a property `index` set to the value of the array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). Currently, tasks are missing the property of `index`.
![image](https://user-images.githubusercontent.com/33268927/202714866-e95753b1-8c23-440e-bfd7-d29d391badec.png)